plugin-review.md

```diff
@@ -1,243 +1,193 @@
 
-Some examples:
+For example, Node.js packages like `fs`, `crypto`, and `os`, are only available on desktop.
 
-- `SubtleCrypto` is the web equivalent of `require('crypto')`.
-- `navigator.clipboard.readText()` and `navigator.clipboard.writeText()` to access clipboard contents.
+If possible, use alternative features that are available in the Web API. For example:
 
-If you must use a NodeJS or Electron API, then you must change the `isDesktopOnly` flag in your `manifest.json` to `true`.
+- [`SubtleCrypto`](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) instead of [`crypto`](https://nodejs.org/api/crypto.html).
+- `navigator.clipboard.readText()` and `navigator.clipboard.writeText()` to access clipboard contents.
 
 ### Commands
 
-#### When to use different callbacks
-
-Obsidian's command API offers multiple ways to register a command: `callback`, `checkCallback`, `editorCallback` and `editorCheckCallback`.
+When you add a command in your plugin, use the appropriate callback type.
 
-Commands may require some condition to be satisfied, such as having a markdown file open, or having some text selected.
+- Use `callback` if the command runs unconditionally.
+- Use `checkCallback` if the command only runs under certain conditions.
 
-#### Callback command
+If the command requires an open and active Markdown editor, use `editorCallback`, or the corresponding `editorCheckCallback`.
 
-If your command can always run, and does not require any condition, then you should use the most simple Command form `callback`.
-
-```ts
-// An example callback that shows a simple notice.
-callback: () => {
-	new Notice('I am alive!');
-}
-```
+## Workspace
 
-#### Editor Callback
+### Avoid accessing `Workspace.activeLeaf` directly
 
-If your command only requires having an active markdown editor open, then you should use the `editorCallback` which will pass the active `Editor` object as well as the `MarkdownView` object to your callback.
+If you want to access the editor in the active view, use `Workspace.getActiveViewOfType()` instead:
 
 ```ts
-// An example editorCallback that shows a notice of the selection content.
-editorCallback: (editor: Editor) => {
-	new Notice(`Your selection is: "${editor.getSelection()}"`);
+const view = app.workspace.getActiveViewOfType(MarkdownView);
+// getActiveViewOfType will return null if the active view is null,
+// or is not of type MarkdownView.
+if (view) {
+  const editor = view.editor;
+  
+  // Do something with editor
 }
 ```
 
-#### Check Callback
+### Avoid managing references to custom views
 
-If your command can only run under certain conditions, then you should use either `checkCallback` or `editorCheckCallback`.
+ Doing so can cause memory leaks or unintended consequences.
+
+**Don't** do this:
 
 ```ts
-// An example checkCallback that only works on Monday.

```

Comment:
I felt the section on callbacks is documenting what the functions do rather than provide a guideline. I think these sections should point out common feedback and then link to proper documentation instead. If there are none, I'm fine with leaving it in as well.

I was tempted to link to [this page](https://marcus.se.net/obsidian-plugin-docs/guides/commands),but figured we only want to cite official sources here.

Reply:
Oh ok that's fine then. I don't mind linking to external docs; we can always swap the links if it stops working.

Also the `checkCallback` tip in your docs also makes the mistake of only performing the check when `checking` is true - the intended use of the `checkCallback` is to perform the check regardless of `checking`. The design of the `checking` flag is so that you can "reuse" some intermediate values of the check process (perhaps you've done some kind of parsing, or the reference to the active view/file) in the actual action of the callback.